syntax = "proto3";

package roles;

//service Permissions {
//    rpc CheckEmailVerifiedPermission(CheckPermissionRequest) returns (CheckPermissionResponse);
//    rpc CheckCustomerPermission(CheckPermissionRequest) returns (CheckPermissionResponse);
//    rpc CheckCourierPermission(CheckPermissionRequest) returns (CheckPermissionResponse);
//    rpc CheckRestaurantManagerPermission(CheckPermissionRequest) returns (CheckPermissionResponse);
//    rpc CheckModeratorPermission(CheckPermissionRequest) returns (CheckPermissionResponse);
//}
//
//message CheckPermissionRequest {
//    string access_token = 1;
//}
//
//message CheckPermissionResponse {
//    bool has_permission = 1;
//    int64 user_id = 2;
//}

service Roles {
    rpc GetUserRole(GetUserRoleRequest) returns (GetUserRoleResponse);
}

enum UserRole {
    CUSTOMER = 0;
    COURIER = 1;
    RESTAURANT_MANAGER = 2;
    MODERATOR = 3;
}

message GetUserRoleRequest {
    string access_token = 1;
}

message GetUserRoleResponse {
    int64 user_id = 1;
    UserRole role = 2;
}